src/main/java/com/hyperwallet/clientsdk/util/HyperwalletApiClient.java

```diff
@@ -146,6 +146,20 @@ private void checkResponseHeader(Response response) {
         }
     }
 
+    protected void checkContentType(final Request request, final Response response) {
+        String requestAccept = request.getHeader("Accept");
+        String responseContentType = response.getHeader("Content-Type");
+        String json = "application/json";
+        String jose = "application/jose+json";
+        if ((StringUtils.isEmpty(requestAccept) && (StringUtils.isEmpty(responseContentType) || !responseContentType.equals(json))) ||
```

Comment:
Hey sorry, this `if` condition is too complicated, can we break down this into two checks? like if encryption is enabled we will check if only for `content-type` and `accept` headers for `application/jose+json` and if not we will only check mime type `application/json`?